ui/src/Explore/CommitLog.tsx

```diff
@@ -61,13 +61,17 @@ const CommitLog = (props: {
   ]);
   const [showCommitDetails, setShowCommitDetails] = useState(false);
   const [commitDetails, setCommitDetails] = useState<CommitMeta | undefined>();
+  const [isRefNotFound, setRefNotFound] = useState(false);
   const location = useLocation();
   const history = useHistory();
   useEffect(() => {
     const logs = async () => {
-      const results = await fetchLog(currentRef);
-      if (results) {
-        setLogList(results);
+      const fetchLogResults = await fetchLog(currentRef);
+      if (fetchLogResults) {
+        setLogList(fetchLogResults);
+        setRefNotFound(false);
+      } else if (fetchLogResults === undefined) {
+        setRefNotFound(true);
```

Comment:
nit: intead of the if-else, maybe just `setRefNotFound(fetchLogResults === undefined)`